### ISSUE TITLE
Enhanced eCommerce for Google Analytics integration for Woocommerce

### ISSUE THREAD

**Issue by RicaNeaga** [google/site-kit-wp#459]
## Feature Description

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

## Changelog entry

*

**Comment by jamesozzie** [comment]
@RicaNeaga Site Kit is at a developer beta stage at present, but I'll mark it as an enhancement for consideration with the team for future releases. Thanks for the input

**Comment by slobodan** [comment]
Great to hear Enhanced Ecommerce is on the radar!

Since Enhanced Ecommerce tracking via GTM typically requires several triggers and tags created in Google Tag Manager, what would be the best way to create them? Should there be a tutorial so users can do it on their own, using GTM interface, or are there any ways to have this automated via Site Kit? I don't see an option for users to edit what's inside a GTM container from Site Kit at the moment (it should stay that way, imo), only to select the container, so a fully automated solution would probably be difficult at this point.

Would it make sense if Site Kit provided a .json file that can be imported into GTM, so Enhanced Ecommerce in a way that is at least semi-automated?

**Comment by nicomollet** [comment]
Interested in this too.
Currently using https://fr.wordpress.org/plugins/duracelltomi-google-tag-manager/ for tracking EEC transactions.

**Comment by slobodan** [comment]
@nicomollet that's what we use as well, for all of our clients. Never had any problems with it until recently when one of the clients started getting transactions tracked multiple times. But even without that issue, I think moving to Site Kit is the way to go if you want to keep things simple.

What do you think about my question? What would your preferred way of setting up the container be?

**Comment by morganhvidt** [comment]
Is this built-in eCommerce tracking still being considered? 😀 It would be nice for WooCommerce & Easy Digital Downloads.

**Comment by jonathannz** [comment]
> @RicaNeaga Site Kit is at a developer beta stage at present, but I'll mark it as an enhancement for consideration with the team for future releases. Thanks for the input

Is there any estimated time-line for enhanced eCommerce tracking in Site Kit? Seems to me like a much needed feature for the many WC sites running WooCommerce (etc.)

**Comment by jonasskafte** [comment]
I agree @jonathannz! Would love to get an update on this as if it's not coming anytime soon, then we'll be looking for an analytics solution elsewhere (although I would love to stay with Site Kit if at all possible!) Cheers!

**Comment by jonasskafte** [comment]
Does anyone know if Site Kit is currently compatible with either of these plugins or will they conflict?:

[Google Tag Manager for WordPress](https://wordpress.org/plugins/duracelltomi-google-tag-manager/)
[WooCommerce Google Analytics](https://woocommerce.com/products/woocommerce-google-analytics/)

Thanks!

**Comment by MiraSm** [comment]
Is this update planned in the near future?

**Comment by jamesozzie** [comment]
@jonathannz @MiraSm We're working on some additional updates at present, including #1728 which you may find of interest, with additional event tracking for WooCommerce. But keep watch of this issue, we'll add any updates to this feature request here. 

@jonasskafte there should be no issues with compatibility. If you encounter any problems you can open a support topic on the WordPress support page and we'd be happy to assist.

**Comment by monogios** [comment]
Does this plugin does properly track e-commerce data or do you suggest keeping WooCommerce Google Analytics Integration installed as well?

**Comment by richard-osborne** [comment]
Adding a +1 for this please as reluctant to have more than 1 plugin installed for various Google site stats.

**Comment by tvdsluijs** [comment]
Would be great to have this! +1

**Comment by madatr** [comment]
.

**Comment by dougwithseismic** [comment]
Is this still an open feature? What can I do to help this along?

**Comment by bethanylang** [comment]
Hi there @dougwithseismic! Yes, this is still on our list for a future enhancement, but we don't currently have a timeline for release.

**Comment by SOYOO974** [comment]
Until the enhanced e-commerce tracking configuration is implemented in Site Kit, could you please provide guidance to configure it with a workaround? Can we use the  WooCommerce Google Analytics Integration or the MonsterInsight plugins so set this up, without having to create new GA properties?

**Comment by bethanylang** [comment]
@SOYOO974 Thanks for getting in touch! Unfortunately there isn't any workaround that can be implemented via Site Kit. I'd recommend reaching out on the support forums for the other plugins that you named here to see if they have any recommendations. You can also try posting to the [Google Analytics Community](https://support.google.com/analytics/community?hl=en) for help.

**Comment by strarsis** [comment]
WooCommerce integration would be great.
Can an extension for WooCommerce be made that extends this plugin?`
Related: https://github.com/google/site-kit-wp/issues/5430

**Comment by rvisima** [comment]
It's been three years. How has this still not been implemented?

**Comment by that-mike-bal** [comment]
Hey @bethanylang - any chance this one will happen in the near-ish future?

**Comment by FilipeOS** [comment]
I don't understand how things like this are not implemented... 4 years

**Comment by stevegiorgi** [comment]
I came across this: https://github.com/google/site-kit-wp/commit/c8317aab45fa41a59b5c420e811cfd08b8a606d8

Is anyone else using https://fr.wordpress.org/plugins/duracelltomi-google-tag-manager/ and seeing the "purchase" event added to the data layer?  I have a "purchase" event in addition to "gtm4wp".

**Comment by cryptomako** [comment]
Any update on this?
A lot of interest and no updates after 4 years.
looks a great tools, but my site is woocommerce , then is useless at this point.

**Comment by bethanylang** [comment]
Hi all! Wanted to update here from the Site Kit team that we are currently planning the addition of new functionality that will allow users to track important events and define conversions, including e-commerce conversion events from WooCommerce. We don't yet have an ETA on when this will be available in Site Kit as it is still early days, but we'll continue to update here as we have more information!

**Comment by marrrmarrr** [comment]
Hi everyone! As we're scoping the e-commerce conversion feature in Site Kit, it would be super helpful if you could share more concretely what functionality exactly would be useful for you in Site Kit -- is it event instrumentation and reporting in particular? If so, what events are you interested in exactly? (We're already looking at the "usual suspects" like "add to cart" and "purchase", hence having a more detailed list might help us add any less popular events). 

Besides event instrumentation, what else would be helpful for you?

**Comment by slobodan** [comment]
> Hi everyone! As we're scoping the e-commerce conversion feature in Site Kit...

Great news! I think handling most of the events outlined [here](https://developers.google.com/analytics/devguides/collection/ga4/ecommerce?client_type=gtag) would be a great start. So:

* view_item_list
* select_item
* view_item
* add_to_cart
* view_cart
* remove_from_cart
* begin_checkout
* add_shipping_info
* add_payment_info
* purchase
* refund (maybe not possible)

And then also allow developers to easily extend this by registering their own events would create a framework that can handle anything.

**Comment by asafm7** [comment]
@marrrmarrr This might also be a good reference:

https://woocommerce.com/document/google-analytics-integration/

**Comment by bgoewert** [comment]
Support for `add_to_cart` (#8553) and `purchase` (#8814) has been added.
